### PR TITLE
[SPARK-42591][SS][DOCS] Add examples of unblocked workloads after SPARK-42376

### PR DESCRIPTION
**Pull request by HeartSaVioR**
### What changes were proposed in this pull request?

This PR proposes to add examples of unblocked workloads after SPARK-42376, which unblocks stream-stream time interval join followed by stateful operator.

### Why are the changes needed?

We'd like to remove the description of limitations which no longer exist, as well as provide some code examples so that users can get some sense how to use the functionality.

### Does this PR introduce _any_ user-facing change?

Yes, documentation change.

### How was this patch tested?

Created a page via SKIP_API=1 bundle exec jekyll serve --watch and confirmed.

Screenshots:

> Scala

스크린샷 2023-03-07 오후 9 39 36

> Java

스크린샷 2023-03-07 오후 9 39 28

> Python

스크린샷 2023-03-07 오후 9 37 03

### CODE REVIEW

**Comment by HeartSaVioR** [comment]
This should be merged after #39931.

**Comment by HeartSaVioR** [comment]
cc. @zsxwing @rangadi @jerrypeng @anishshri-db @chaoqin-li1123 
cc-ing folks who reviewed the code change PR. This PR is a doc change to show up what is being unblocked, like we did in https://github.com/apache/spark/pull/40188 for fixing broken late record filtering.

**Comment by HeartSaVioR** [comment]
cc. @viirya as well who may be interested with new feature in SS.

File: docs/structured-streaming-programming-guide.md [hunk 1]
```diff
@@ -1848,12 +1848,137 @@ Additional details on supported joins:
 
 - As of Spark 2.4, you can use joins only when the query is in Append output mode. Other output modes are not yet supported.
 
-- As of Spark 2.4, you cannot use other non-map-like operations before joins. Here are a few examples of
-  what cannot be used.
+- You cannot use mapGroupsWithState and flatMapGroupsWithState before and after joins.
 
-  - Cannot use streaming aggregations before joins.
+In append output mode, you can construct a query having non-map-like operations before/after join.
```
Comment (viirya) [inline docs/structured-streaming-programming-guide.md:1853]: `non-map-like operations` sounds not clear to me. Can you add some examples like:

```
In append output mode, you can construct a query having non-map-like operations, e.g. ..., ..., ..., before/after join.
```

File: docs/structured-streaming-programming-guide.md [hunk 2]
```diff
@@ -1848,12 +1848,137 @@ Additional details on supported joins:
...
+    clickTime >= impressionTime AND
+    clickTime <= impressionTime + interval 1 hour
+  """),
+  "leftOuter"                 # can be "inner", "leftOuter", "rightOuter", "fullOuter", "leftSemi"
+)
+
+joined.groupBy(
+  joined.clickAdId,
+  window(joined.clickTime, "1 hour")
+).count()
```
Comment (viirya) [inline docs/structured-streaming-programming-guide.md:1909]: This python example seems duplicate to the bottom one (join followed by time window aggregation).

File: docs/structured-streaming-programming-guide.md (hunk 1 above)
Comment (HeartSaVioR) [reply docs/structured-streaming-programming-guide.md:1853]: Sounds better. I'll list some operations here.

File: docs/structured-streaming-programming-guide.md (hunk 2 above)
Comment (HeartSaVioR) [reply docs/structured-streaming-programming-guide.md:1909]: Looks like something is messed up. Nice finding.

**Review by viirya — APPROVED** [review]

**Comment by HeartSaVioR** [comment]
Thanks for reviewing! Merging to master.

**Pull request closed without merging**